server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java

```diff
@@ -624,11 +654,31 @@ public MemoryUsage overLimit(MemoryUsage memoryUsed) {
                         blackHole += localBlackHole;
                         logger.trace("black hole [{}]", blackHole);
 
-                        long now = timeSupplier.getAsLong();
-                        this.lastCheckTime = now;
-                        allocationDuration = now - begin;
+                        this.lastCheckTime = timeSupplier.getAsLong();
                         this.attemptNo = 0;
                     }
+
+                    if (gcTriggeredSuccessfully == false) {
```

Comment:
This consider both conditions before - if the GC count was increased, even if it did not bring memory down, we are not doing a full GC. Should we concentrate just on memory?

Reply:
Updated to test memory reclaimed independently before triggering full GC